ts/WoltLabSuite/Core/Controller/User/Profile.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Handles user profile functionalities.
+ *
+ * @author Marcel Werk
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module WoltLabSuite/Core/Controller/User/Profile
+ */
+
+import { UserList } from "../../Component/User/List";
+
+function setupUserList(userId: number, buttonId: string, className: string): void {
+  const button = document.getElementById(buttonId) as HTMLElement;
+  if (button) {
+    let userList: UserList;
+
+    button.addEventListener("click", () => {
+      if (userList === undefined) {
+        userList = new UserList(
+          {
+            className: className,
+            parameters: {
+              userID: userId,
+            },
+          },
+          button.closest("section")!.querySelector(".boxTitle")!.textContent!.replace(/ \d+$/, ""),
```

Comment:
The replacement regex makes quite a few assumptions about the HTML layout. Should a clearly identifiable `<span>` or something like that be added?

Reply:
This is not possible, because the title is taken from the box title which is unknown in JavaScript. The only source is the generated HTML which at least is directly based off the `__box` template so that we can at least assume the HTML to be matching. However, it might be better to use `.closest(".box")` instead of `.closest("section")`.

Reply:
This module is specific to the user profile, can't `userSidebar.tpl` be modified from:

https://github.com/WoltLab/WCF/blob/09aab20f27805dca11b564d613d5c30bdb17ccc9/com.woltlab.wcf/templates/userSidebar.tpl#L4

to:

```
<h2 class="boxTitle"><span class="this_is_the_real_title">{lang}wcf.user.profile.following{/lang}</span> <span class="badge">{#$followingCount}</span></h2>
```

or similar? I'm mostly concerned about the number replacement, which is super ugly.